server/src/main/java/org/elasticsearch/action/ingest/GetPipelineResponse.java

```diff
@@ -122,4 +120,43 @@ public static GetPipelineResponse fromXContent(XContentParser parser) throws IOE
         ensureExpectedToken(XContentParser.Token.END_OBJECT, parser.currentToken(), parser::getTokenLocation);
         return new GetPipelineResponse(pipelines);
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == null) {
+            return false;
+        } else if (other instanceof GetPipelineResponse){
+            GetPipelineResponse otherResponse = (GetPipelineResponse)other;
+            if (pipelines == null) {
+                return otherResponse.pipelines == null;
+            } else {
+                // We need a map here because order does not matter for equality
+                Map<String, PipelineConfiguration> otherPipelineMap = new HashMap<>();
+                for (PipelineConfiguration pipeline: otherResponse.pipelines) {
+                    otherPipelineMap.put(pipeline.getId(), pipeline);
+                }
+                for (PipelineConfiguration pipeline: pipelines) {
+                    PipelineConfiguration otherPipeline = otherPipelineMap.get(pipeline.getId());
+                    if (otherPipeline == null ||
+                        !pipeline.getConfigAsMap().equals(otherPipeline.getConfigAsMap())) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 1;
+        for (PipelineConfiguration pipeline: pipelines) {
+            // We only take the sum here to ensure that the order does not matter.
+            result += (pipeline == null ? 0 : pipeline.getConfigAsMap().hashCode());
```

Comment:
shall we add equals and hashcode to PipelineConfiguration ?

Reply:
It is already there. Unfortunately, it uses the `BytesReference` to do this. This depends on the `XContentType`. Do you want me to change the implementation there?

Reply:
I think we could, I am assuming that we need these only for tests and making equals/hashcode work with the map representation would not hurt. You can try and run `IngestMetadaTests`, `PipelineConfigurationTests`, `PipelineExecutionServiceTests` and `PipelineStoreTests` to check whether they remain green after the change.

Reply:
Cool thanks! Will do the change.